lib/model/query/forms.js

```diff
@@ -329,15 +329,27 @@ const publish = (form) => async ({ Forms, Datasets }) => {
     : {};
 
   const publishedAt = (new Date()).toISOString();
-  return Promise.all([
+  await Promise.all([
     Forms._update(form, { currentDefId: form.draftDefId, draftDefId: null, ...enketoIds }),
     Forms._updateDef(form.def, { draftToken: null, enketoId: null, publishedAt }),
-    Datasets.publishIfExists(form.def.id, publishedAt)
   ])
     .catch(Problem.translate(
       Problem.user.uniquenessViolation,
       () => Problem.user.versionUniquenessViolation({ xmlFormId: form.xmlFormId, version: form.def.version })
     ));
+
+  const ds = await Datasets.publishIfExists(form.def.id, publishedAt);
+  // this check is for issue c#554. we will hopefully come back to this flow and improve it later.
+  // ds contains a list of objects about what happened with the dataset
+  // like if the dataset was published and which properties were published.
+  // if it's empty, there is no dataset to work with.
+  if (!ds || ds.length > 0) {
+    const dataset = await Datasets.getById(ds[0].id).then(o => o.get());
+    const attachment = await FormAttachments.getByFormDefIdAndName(form.def.id, `${dataset.name}.csv`);
```

Comment:
CSV file extension should be case insensitive. This might be an issue in the standard linking case (where the dataset already exists).

Reply:
confirmed that this is an issue in the other auto link path as well